dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRestRepositoryIT.java

```diff
@@ -1793,6 +1851,797 @@ public void getDirectEpersonGroups() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.delete(context, registrationDataService.findByEmail(context, ePerson.getEmail()));
+        registrationDataService.deleteByToken(context, tokenForEPerson);
+        context.restoreAuthSystemState();
+
+    }
+
+    @Test
+    public void registerNewAccountPatchUpdatePasswordRandomUserUuidFail() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String newRegisterEmail = "new-register@fake-email.com";
+        RegistrationRest registrationRest = new RegistrationRest();
+        registrationRest.setEmail(newRegisterEmail);
+        getClient().perform(post("/api/eperson/registrations")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(mapper.writeValueAsBytes(registrationRest)))
+                   .andExpect(status().isCreated());
+
+        EPerson ePerson = EPersonBuilder.createEPerson(context)
+                                        .withNameInMetadata("John", "Doe")
+                                        .withEmail("Johndoe@fake-email.com")
+                                        .withPassword(password)
+                                        .build();
+
+        String newPassword = "newpassword";
+
+        context.restoreAuthSystemState();
+
+        List<Operation> ops = new ArrayList<Operation>();
+        ReplaceOperation replaceOperation = new ReplaceOperation("/password", newPassword);
+        ops.add(replaceOperation);
+        String patchBody = getPatchContent(ops);
+        accountService.sendRegistrationInfo(context, ePerson.getEmail());
+        String newRegisterToken = registrationDataService.findByEmail(context, newRegisterEmail).getToken();
+        PasswordHash oldPassword = ePersonService.getPasswordHash(ePerson);
+        // updates password
+        getClient().perform(patch("/api/eperson/epersons/" + ePerson.getID())
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON)
+                                     .param("token", newRegisterToken))
+                        .andExpect(status().isUnauthorized());
+
+        PasswordHash newPasswordHash = ePersonService.getPasswordHash(ePerson);
+        assertTrue(StringUtils.equalsIgnoreCase(oldPassword.getHashString(),newPasswordHash.getHashString()));
+        assertFalse(registrationDataService.findByEmail(context, ePerson.getEmail()) == null);
+        assertFalse(registrationDataService.findByEmail(context, newRegisterEmail) == null);
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.delete(context, registrationDataService.findByEmail(context, ePerson.getEmail()));
+        registrationDataService.deleteByToken(context, newRegisterToken);
+        context.restoreAuthSystemState();
```

Comment:
this should go in a finally to be sure that it is executed also in case of test failure, otherwise a failure in this test due to a refactoring would produce a lot of unrelated failure due to a dirty testenv

Reply:
We'll move this to a finally